### PR TITLE
Setup CI/CD C++ workflow

### PR DESCRIPTION
**Pull request by wrobcio789**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+cmake_minimum_required(VERSION 3.19)
+project(ProGen)
+
+set(CMAKE_CXX_STANDARD 20)
+
+add_executable(ProGen src/main.cpp)
```
Comment (jakpiase) [inline CMakeLists.txt]: Please add newline at EOF

Comment (wrobcio789) [reply CMakeLists.txt]: Haven't requested the review yet. Don't be so eager

**Comment by wrobcio789** [comment]
Verify only cmake files, and .github dir + sample test. I took commits from another PR just to have some code to work with. Won't be setting up linter for now, cause its not that easy. I recommend installing and using clang tidy locally.

Since I'd like commits in your PR to be squashed, let's merge your PR first, then mine. I'll resolve any conflicts.

**Comment by wrobcio789** [comment]
Do the review bro

File: .github/workflows/c-cpp.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    - name: build
+      working-directory: ${{runner.workspace}}/build
+      run: cmake --build .
+
+    - name: test
+      working-directory: ${{runner.workspace}}/build/tests
+      run: ctest --verbose
+      env:
+        GTEST_COLOR: 1
```
Comment (jakpiase) [inline .github/workflows/c-cpp.yml]: please, add newline at EOF

File: tests/algorithms/sample_algorithm_test.cpp [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#include "gtest/gtest.h"
+
+TEST(Sample_tests, addition_test) {
+    int addend_a = 2;
+    int addend_b = 3;
+
+    int sum = addend_a + addend_b;
+
+    EXPECT_EQ (sum,  5);
+}
```
Comment (jakpiase) [inline tests/algorithms/sample_algorithm_test.cpp]: please, add newline at EOF

File: tests/main.cpp [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#include "gtest/gtest.h"
+
+int main(int argc, char **argv) {
+    ::testing::InitGoogleTest(&argc, argv);
+
+    return RUN_ALL_TESTS();
+}
```
Comment (jakpiase) [inline tests/main.cpp]: please, add newline at EOF

File: tests/algorithms/sample_algorithm_test.cpp [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#include "gtest/gtest.h"
+
+TEST(Sample_tests, addition_test) {
+    int addend_a = 2;
```
Comment (jakpiase) [inline tests/algorithms/sample_algorithm_test.cpp:4]: Did you mean "added"? Why not just "a"?

Comment (wrobcio789) [reply tests/algorithms/sample_algorithm_test.cpp:4]: Google addend word bro

Comment (jakpiase) [reply tests/algorithms/sample_algorithm_test.cpp:4]: Still, if you want to be precise why didn't you name it as "addend_int_variable_a_equals_two"?

**Pull request merged**